deployment/pom.xml

```diff
@@ -25,6 +25,10 @@
       <groupId>io.quarkus</groupId>
       <artifactId>quarkus-rest-client-jackson-deployment</artifactId>
     </dependency>
+    <dependency>
+      <groupId>io.quarkus</groupId>
+      <artifactId>quarkus-oidc-client-filter-deployment</artifactId>
```

Comment:
@ricardozanini  for the record, this leaves the extension prepared to work
1) with the client filter by not having to add any additional dependencies on the users
2) work well in native mode

Reply:
But it does bring the dependency if not needed in scenarios where users won't use OAuth2?

Reply:
somehow yes, but I believe it's a fair trade-off between letting the extension be more "self-contained" or shift some configuration aspects to the user. Also, when we go to native I'm not sure if we could introduce other issues.